phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/ui/LeftListWidgetChooser.java

```diff
@@ -48,27 +57,54 @@ public LeftListWidgetChooser(List<CombinedListWidget<E>> listWidgets) {
         this.iconsListButton.setToggleGroup(toggleGroup);
         this.iconsListButton.getStyleClass().addAll("listIcon", "iconsList");
         this.iconsListButton
-                .setOnAction(event -> listWidgets.forEach(widget -> widget.showList(ListWidgetType.ICONS_LIST)));
+                .setOnAction(event -> {
+                    listWidgets.forEach(widget -> widget.showList(ListWidgetType.ICONS_LIST));
+                    this.onChoose.ifPresent(consumer -> consumer.accept(ListWidgetType.ICONS_LIST));
+                });
         this.iconsListButton.addEventFilter(ActionEvent.ANY, filter);
 
         this.compactListButton = new ToggleButton();
         this.compactListButton.setToggleGroup(toggleGroup);
         this.compactListButton.getStyleClass().addAll("listIcon", "compactList");
         this.compactListButton
-                .setOnAction(event -> listWidgets.forEach(widget -> widget.showList(ListWidgetType.COMPACT_LIST)));
+                .setOnAction(event -> listWidgets.forEach(widget -> {
+                    widget.showList(ListWidgetType.COMPACT_LIST);
+                    this.onChoose.ifPresent(consumer -> consumer.accept(ListWidgetType.COMPACT_LIST));
```

Comment:
Watch out, you're adding the `onChoose` call to the forEach loop. This means you will call it three times!